### PR TITLE
feat: Update implicit OpenID attribute criteria and extend e2e tests.

### PR DESCRIPTION
**Pull request by sea-snake**
Extend direct OpenID attribute e2e tests.
# Changes
- Implicitly allow incoming attribute requests based on exact matches instead of prefix.
- Update test OpenID provider implementation to support arbitrary port.
- Update test workflow to start two OpenID instances with port 11105 and 11106.
- Update OpenID fixture to support creating OpenID users for arbitrary OpenID instances.
- Implement authorization fixture for e2e tests that involve authorization through II
   - Add configuration to define protocol and requested attributes (if supported by protocol).
   - Returns authorized principal and attributes fixtures.
- Implement OpenID sign-up tests to use above updated OpenID fixture.
- Extend direct OpenID tests to use above authorization and OpenID fixtures to test all edge cases.

### CODE REVIEW

File: src/frontend/src/routes/(new-styling)/(resuming-channel)/resume-openid-authorize/+page.svelte [hunk 1]
```diff
@@ -51,7 +51,9 @@
         return;
       }
       const implicitConsentAttributeKeys = paramsResult.data.attributes.filter(
-        (attribute) => attribute.startsWith(`openid:${issuer}:`),
+        (attribute) =>
+          attribute === `openid:${issuer}:name` ||
+          attribute === `openid:${issuer}:email`,
```
Comment (aterga) [inline src/frontend/src/routes/(new-styling)/(resuming-channel)/resume-openid-authorize/+page.svelte:56]: PR title should be marked as `feat: ` and should explain that this isn't a test-only PR

File: src/frontend/tests/e2e-playwright/authorize/openid.spec.ts [hunk 2]
```diff
@@ -1,48 +1,146 @@
...
+    test.beforeEach(
+      async ({ page, identity, signInWithOpenId, openIdUsers }) => {
+        await page.goto(II_URL + "/manage/access");
+        await identity.signIn();
+
+        for (const user of openIdUsers) {
+          // Click "Add new" and pick OpenID provider
+          await page.getByRole("button", { name: "Add new" }).click();
+          const popupPromise = page.context().waitForEvent("page");
+          await page.getByRole("button", { name: user.issuer.name }).click();
```
[the remark this answers is not in the document]
Comment (sea-snake) [reply src/frontend/tests/e2e-playwright/authorize/openid.spec.ts:120]: This is intentionally a partial match to ensure the matcher doesn't break when the button name is slightly renamed later on.

File: src/frontend/src/routes/(new-styling)/(resuming-channel)/resume-openid-authorize/+page.svelte (hunk 1 above)
Comment (sea-snake) [reply src/frontend/src/routes/(new-styling)/(resuming-channel)/resume-openid-authorize/+page.svelte:56]: Good catch!

File: src/frontend/tests/e2e-playwright/index.spec.ts [hunk 3]
```diff
@@ -147,93 +146,93 @@ test.describe("First visit", () => {
...
+      page,
+      signInWithOpenId,
+      openIdUsers,
+    }) => {
+      // Pick OpenID to continue
+      await page.goto(II_URL);
+      await page.getByRole("button", { name: "Sign in" }).click();
+      const popupPromise = page.context().waitForEvent("page");
+      await page
+        .getByRole("button", { name: openIdUsers[0].issuer.name })
```
[the remark this answers is not in the document]
Comment (sea-snake) [reply src/frontend/tests/e2e-playwright/index.spec.ts:174]: This is intentionally a partial match to ensure the matcher doesn't break when the button name is slightly renamed later on.

File: src/frontend/tests/e2e-playwright/authorize/index.spec.ts [hunk 4]
```diff
@@ -169,28 +174,16 @@ test("Authorize by signing up with OpenID", async ({ page }) => {
     // Pick OpenID to continue
     const pagePromise = authPage.context().waitForEvent("page");
     await authPage
-      .getByRole("button", { name: "Continue with Test OpenID" })
+      .getByRole("button", {
+        name: openIdUsers[0].issuer.name,
```
[the remark this answers is not in the document]
Comment (sea-snake) [reply src/frontend/tests/e2e-playwright/authorize/index.spec.ts:178]: This is intentionally a partial match to ensure the matcher doesn't break when the button name is slightly renamed later on.
